public/app/plugins/datasource/loki/querybuilder/components/QueryPatternsCard.tsx

```diff
@@ -0,0 +1,109 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, Card, useStyles2 } from '@grafana/ui';
+import { RawQuery } from 'app/plugins/datasource/prometheus/querybuilder/shared/RawQuery';
+
+import logqlGrammar from '../../syntax';
+import { lokiQueryModeller } from '../LokiQueryModeller';
+import { LokiQueryPattern } from '../types';
+
+type Props = {
+  pattern: LokiQueryPattern;
+  hasNewQueryOption: boolean;
+  hasPreviousQuery: boolean;
+  selectedPatternName: string | null;
+  setSelectedPatternName: (name: string | null) => void;
+  onPatternSelect: (pattern: LokiQueryPattern, selectAsNewQuery?: boolean) => void;
+};
+
+export const QueryPatternsCard = (props: Props) => {
+  const { pattern, onPatternSelect, hasNewQueryOption, hasPreviousQuery, selectedPatternName, setSelectedPatternName } =
+    props;
+
+  const styles = useStyles2(getStyles);
+  const lang = { grammar: logqlGrammar, name: 'logql' };
+
+  return (
+    <Card key={pattern.name} className={styles.card}>
```

Comment:
no need to set the key here again.

Reply:
correct! I missed this when I refactored the solution to use QueryPatternsCards.